src/include/sof/compiler_info.h

```diff
@@ -0,0 +1,55 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Karol Trzcinski <karolx.trzcinski@linux.intel.com>
+ */
+
+/**
+ * \file include/sof/compiler_info.h
+ * \brief Compiler version and name descriptor
+ * \author Karol Trzcinski <karolx.trzcinski@linux.intel.com>
+ */
+
+#ifndef __COMPILER_INFO_H__
+#define __COMPILER_INFO_H__
+
+#include "ipc/info.h"
+
+/* read used compilator name and version */
+/* CC_NAME must consist of 3 characters with null termination */
+/* See declaration of sof_ipc_cc_version. */
+#if defined(__XCC__)
+#define CC_MAJOR (XTHAL_RELEASE_MAJOR / 1000)
+#define CC_MINOR ((XTHAL_RELEASE_MAJOR % 1000) / 10)
+#define CC_MICRO XTHAL_RELEASE_MINOR
+#define CC_NAME "XCC"
+#if defined(XCC_TOOLS_VERSION)
```

Comment:
There should not be dependency from C code to .sh utility script. You can inject define from buildsystem.
BTW cmake -Dsomething will not propagate define down to C, it is visible only i cmake files ;)

Reply:
Ok, done